\documentclass{article}

%\bibpunct{(}{)}{;}{a}{}{,}
\usepackage{natbib}
\usepackage{palatino}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
%\usepackage[usenames,dvipsnames]{xcolor} %% for non-standard cite color
\usepackage{graphicx}
\usepackage[colorlinks = true, linkcolor = blue, urlcolor=blue,citecolor = MidnightBlue]{hyperref}
\usepackage{url}
\usepackage{enumerate}
%\usepackage[normalem]{ulem}
\usepackage[left=50pt,
            textwidth=345pt,
            marginparsep=25pt,
            marginparwidth=124pt]{geometry}
%\usepackage[margin=1in]{geometry}

\newcommand{\etal}{et al.}
\newcommand{\shir}{Shirreff et al.}%{\citeauthor{shirreff_transmission_2011}}
\newcommand{\champr}{Champredon et al.}%{\citeauthor{champredon_hiv_2013}}
\newcommand{\tsub}[2]{#1_{{\small \textrm{#2}}}}
\newcommand{\rzero}{{\cal R}_0}
\usepackage{changes}
\usepackage[backgroundcolor=lightgray,textsize=small]{todonotes}

<<setup,echo=FALSE>>=
library("knitr")
opts_chunk$set(echo=FALSE,error=FALSE,fig.width=8,fig.height=4) #
## https://github.com/yihui/tikzDevice/issues/46 
local({
    hook_document <- knitr::knit_hooks$get('document')
    knitr::knit_hooks$set(document = function(x){
        res <- hook_document(x)
        gsub('\\usepackage[]{color}', '\\usepackage[usenames,dvipsnames]{xcolor}', res, fixed = TRUE)
  })
})
@


<<pkgs,warning=FALSE,message=FALSE>>=
library("ggplot2"); theme_set(theme_classic())
library("MASS")
## use Dark2 rather than Set1 because colour #6 of Set1 is yellow (ugh/too light)
scale_colour_discrete <- function(...,palette="Dark2") scale_colour_brewer(...,palette=palette)
scale_fill_discrete <- function(...,palette="Dark2") scale_fill_brewer(...,palette=palette)
library("gridExtra")
library("plyr")  ## for ldply()
library("dplyr") ## for full_join()
library("reshape2")  ## for melt()
library("GGally")
library("knitr")
## devtools::install_github("lionel-/ggstance")
library("ggstance")
@

<<loadstuff,cache=TRUE>>=
load("../simdata/combine_ev_LHS4.rda")
@ 

\author{Sang Woo Park and Ben Bolker}
\title{Effects of epidemiological structure on the transient evolution of HIV virulence}

\begin{document}
\maketitle

\abstract{The evolutionary dynamics of parasite virulence over the
  course of an emerging epidemic have important implications both for
  our basic understanding of epidemiological dynamics and,
  potentially, for the outcomes of public health interventions. In
  general changes in fitness landscapes over the course of an epidemic
  will select for higher virulence during the early,
  exponential-growth phase of the epidemic, but quantitative outcomes
  can depend sensitively on biological details and the structure of
  mathematical models used to capture them.  Fraser, Shirreff, and
  co-workers have proposed a series of models for eco-evolutionary
  dynamics of HIV that are relatively detailed in their portrayal of
  the tradeoffs between transmission and virulence (mediated by
  set-point viral load, SPVL) and their heritability between
  hosts. However, these models use very simple implicit
  representations of the transmission process that ignore the
  partnership dynamics that previous research has found to be critical
  in predicting epidemics of sexually transmitted diseases.  We
  explore models that combine HIV virulence tradeoffs with a range of
  epidemiological structures, modeling partnership formation and
  dissolution and allowing for individuals to transmit disease outside
  of partnerships. We assess summary statistics such as the peak value
  of virulence (SPVL) and the time at which the peak occurs across all
  models and across a Latin hypercube sample that captures a realistic
  range of partnership dynamic parameters for sub-Saharan Africa. In
  order to account for the different interpretations of parameters
  across model structures, we scale all parameter sets to constrain
  the simulated epidemic growth rate to be identical, matching a
  realistic baseline value. Our primary result is that, for this
  particular model setting, the simplest random-mixing structure is
  actually the best approximation to the most realistic model; this
  surprising outcome occurs because the dominance of extra-pair
  contact in the realistic model tends to mask the effects of
  partnership structure.
}
\section{Introduction}

The evolution of pathogen virulence is a fundamental process in
evolutionary biology, of both theoretical and (potentially) practical
importance. The trade-off theory \citep{Ebert1999} --- which
postulates that parasite virulence can be explained as the long-term
evolutionary outcome of a saturating relationship between parasite
clearance rate and transmission rate --- has been criticized
\citep{EbertBull2003,alizon_adaptive_2015}, but has also been
successfully applied in a variety of host-pathogen systems \citep{Dwyer+1990,mackinnon1999genetic,jensen2006empirical,deroode2008virulence}. One
particularly interesting application of these ideas is the work by
Fraser \etal\ showing that HIV appears to satisfy the prerequisites of
the tradeoff theory: in a study of discordant couples (i.e. long-term
sexual partnerships with one infected and one uninfected partner), HIV
virulence as measured by the rate of progression to AIDS was both
heritable and covaried with the set-point viral load (i.e., the
characteristic virus load measured in blood during the intermediate
stage of infection), which in turn predicted the probability of
within-couple transmission
\citep{Fraser+2007,fraser_virulence_2014}\todo{read/check fraser+
  2014!; check vs. \cite{herbeck_is_2012}}. Subsequent studies
\citep{shirreff_transmission_2011,herbeck_hiv_2014} used these data to
parameterize mechanistic models of HIV virulence evolution, suggesting
that HIV invading a novel population would initially evolve increased
virulence, peaking after approximately [XXX] years and then declining
slightly to a long-stable virulence level.

\todo{\cite{gras2009viral} and \cite{muller2009increasing} say that mean virulence has been increasing in Netherlands and Italy. Empirical support? Worth checking out maybe? I think these are similar to what \cite{herbeck_is_2012} says about the evolutionary trend but more specific...}

The work of \shir, and particularly the predicted
transient peak in HIV virulence midway through the epidemic,
highlights the importance of interactions between epidemiological and
evolutionary factors \citep{day_virulence_2004,alizon_price_2009}.
However, despite the attention to mechanistic detail at the individual
or physiological level, the epidemiological structures used in these
models are relatively simple. \todo{don't forget to check other recent papers: esp \cite{payne_impact_2014}}

\todo{\cite{lythgoe2013hiv} --- multi-strain model + within host dynamics but assumes random mixing. This one considers multi-strains existing within a host. Another complication that is important but this paper also ignores the partnership dynamics...}

\todo{\cite{van2014immuno} --- seems even more complicated but doesn't seem like it mentions anything about partnership dynamics... HIV phases are based on Fraser and Hollingsworth et al} 

\todo{\cite{herbeck2016evolution} --- seems better than the previous herbeck et al paper but still weird and is not published (bioRxiv)..} As we discuss in detail below, the
existing models of HIV eco-evolutionary dynamics either use implicit
models that incorporate the average effects of within-couple sexual
contact --- without representing the explicit dynamics of pair
formation and dissolution or accounting for extra-partnership contact
--- or use an agent-based formulation with parameters that effectively
lead to random mixing among infected and uninfected individuals. Here
we explore the effects of incorporating \emph{explicit}
epidemiological structure in eco-evolutionary models.

We add complexity to the epidemiological model following the general approach of \cite{champredon_hiv_2013}; individuals join and leave partnerships at a specified rate, and can have sexual contact both within and outside of established partnerships. At the same time, our analysis somewhat simplifies the models of \shir, for computational tractability; we check that our qualitative results are not sensitive to these simplifications. In order to explore how virulence evolution depends on epidemiological structure, we consider a series of models with increasing levels of complexity. In order to avoid dependence of the results on a particular set of parameters --- as we explain below, finding matching sets of parameters across models with widely differing epidemiological structures is challenging --- we evaluate our models across a wide range of parameters, again following \champr\ in using a Latin hypercube design. For each model run, we compute a set of metrics (peak virulence, timing of virulence peak, equilibrium virulence) that summarize the evolutionary trajectory of a simulated HIV epidemic.

\section{Methods}

\todo{more papers on SPVL heritability (\emph{maybe} include): \cite{alizon_phylogenetic_2010,leventhal_potential_2016}}

As our primary goal is to explore how different epidemiological structures (i.e. partnership dynamics and contact structures) affect our conclusions about the evolution of virulence, our models use a simplified description of within-host dynamics and heritability derived from 
\shir's multi-strain evolutionary model. Like \shir, we use a simple susceptible-infected-susceptible demographic formulation; rather than modeling birth and death (or more specifically, recruitment into the sexually active population and death), we assume that whenever an individual dies from infection, another enters the susceptible compartment.

\subsubsection{Infection dynamics}

Like \shir, we focus on the evolution of mean $\log_{10}$ set-point viral load, SPVL (which we denote as $\alpha$), rather than virulence (i.e. rate of progression to AIDS) itself. 
In contrast to \shir, we use a single-stage disease model instead of accounting explicitly for progression through the three main stages of HIV infection (primary, asymptomatic, and disease), and we use a simple exponentially distributed infectious period instead of a more realistic Weibull-distributed infectious period. We account for varying transmission rates and durations of each disease stage by summing the durations of three stages (again based on \shir's model) and taking the duration-weighted average of transmission rates of three stages. Thus the within-couple transmission rate, $\beta$, for our models is given by:
\begin{equation}
\beta (\alpha) = \frac{D_P \beta_P + D_A (\alpha) \beta_A (\alpha) + D_D \beta_D}{D_P + D_A (\alpha) + D_D}
\end{equation}
where the duration of infection ($D_P$ and $D_D$) and rate of transmission ($\beta_P$ and $\beta_D$) of the \textbf{P}rimary and \textbf{D}isease stages
of infection are independent of the host's SPVL. Following \shir, the duration of infection ($D_A$) and rate of transmission ($\beta_A$) for the \textbf{A}symptomatic stage are Hill functions of the SPVL:

\begin{equation}
\begin{split}
D_A(\alpha) &= \frac{\tsub{D}{max} D_{50} ^{D_k}}{V_\alpha ^{D_k} + D_{50}^{D_k}}, \\
\beta_A(\alpha) &= \frac{\tsub{\beta}{max} V_\alpha ^ {\beta_k}}%
{V_\alpha^{\beta_k} + \beta_{50} ^{\beta_k}},
\end{split}
\end{equation}
where $V_{\alpha} = 10^\alpha$. 
The \textbf{u}ncoupled and \textbf{e}xtra-couple transmission rates are scaled by multiplying the \textbf{w}ithin-couple transmission rate $\beta$ by the contact ratios $c_u/c_w$ and $c_e/c_w$.

\subsubsection{Mutation}

Like \shir\ we incorporate a between-host mutation process in the SPVL, but simplify \shir's evolutionary model slightly by using a one-to-one genotype-phenotype mapping. 
The mutational process in our model is directly taken from \shir. Over the course of infection, mutation occurs within the host. However, it is assumed that SPVL of an infected individual is determined by the SPVL at the time of infection for simplicity (and is not further affected by within-host mutation). Instead, the mutational effect takes place when an infected individual transmits the virus to a susceptible individual. First, the distribution of $\log_{10}$ SPVL is discretized into a vector:

\begin{equation}
\alpha_i =  (\tsub{\alpha}{max} - \tsub{\alpha}{min})\frac{(i-1)}{n-1} + \tsub{\alpha}{min} \qquad i = 1,2,3, \dots n.
\end{equation}
Then, we construct an $n$ by $n$ mutational matrix, $M$ --- which is multiplied with the transmission term ---  so that $M_{ij}$ is the probability that a newly infected individual will have $\log_{10}$ SPVL of $\alpha_j$ given that the infector has $\log_{10}$ SPVL of $\alpha_i$. Finally, the probabilities are normalized so that each row sums to 1:

\begin{equation}
M_{ij} = \frac{\Phi(\alpha_j + d/2;i) - \Phi(\alpha_j - d/2;i)}{\Phi(\tsub{\alpha}{max} + d/2;i) - \Phi(\tsub{alpha}{min} - d/2;i)},
\end{equation}

where $\Phi(x;i)$ is the Gaussian cumulative distribution function with mean $\alpha_i$ and variance of $\sigma_M^2$, and $d = (\tsub{\alpha}{max} - \tsub{\alpha}{min})/(n-1)$. Transmission rate and disease induced mortality rates are discretized into a vector as well:

\begin{equation}
\begin{aligned}
\beta_i &= \beta(\alpha_i)\\
\lambda_i &= \frac{1}{D_P + D_A (\alpha_i) + D_D}
\end{aligned}
\end{equation}


\subsubsection{Contact structure and partnership dynamics}

We developed six multi-strain evolutionary models, designed to cover a gamut between \champr's relatively realistic and \shir's relatively simplistic epidemiological structures, each of which is based on different assumptions regarding contact structure and partnership dynamics. Specifically, we focus on the effects of the assumptions of (1) instantaneous vs. non-instantaneous partnership formation and (2) zero vs. positive extra-partnership sexual contact and transmission on the evolution of mean $\log_{10}$ SPVL.

Our first four models consider explicit partnership dynamics and are based on \champr's model. Model 1 and 2 assume non-instantaneous partnership formation (i.e. individuals spend some time uncoupled, outside of partnerships) and consist of five states that are classified by infection status and partnership status. $S$ is the number of single (uncoupled) susceptible individuals, and $I$ is the number of single infected individuals. $SS$ is the number of susceptible-susceptible couples, $SI$ is the number of serodiscordant (susceptible-infected) couples, and $II$ is the number of concordant positive (infected-infected) couples. Model 1 includes extra-partnership contact (with both uncoupled individuals and individuals in other partnerships) whereas model 2 only considers within-couple transmission. Models 3 and 4 assume instantaneous partnership formation and thus consist of only the three partnered states: $SS$, $SI$, and $II$. Parallel to model 1 and 2, model 3 includes extra-partnership contact (now only with individuals in other partnerships, since uncoupled individuals don't exist in this model) and model 4 only considers within-couple transmission.

In contrast, models 5 and 6 are not explicitly structured.  Model 5 is an implicit serial monogamy model based on the epidemiological model used by \shir. It is actually a random mixing model that consist of only two states, $S$ and $I$, and does not consider explicit partnership dynamics. However, to simulate the effect of instantaneous partnership formation, it uses an adjusted transmission rate that is derived from approximated basic reproduction number of a serial monogamy model \citep{hollingsworth_hiv1_2008}. Finally, model 6 is a simple random-mixing model.

The base model (i.e. model 1) for the first four models is an extension of \champr's model. Individuals in single compartment acquire a partner at a rate, $\rho$, and partnerships dissolve at a rate, $c$. Infected individuals in a discordant partnership infect susceptible partner at a rate $\beta$ (within-couple transmission rate) and susceptible individuals outside the partnership at a rate $c_e$ (extra-couple transmission rate). Likewise, a single infected individual can infect any susceptible individuals at a rate $c_u$ through uncoupled mixing. Extra-couple and uncoupled transmission are modeled in a same way as \champr's model. All the details have been adapted to a multi-strain scenario. Model 2, 3, and 4 are derived from the base model by removing epidemiological details (partnership formation and uncoupled/extra-couple contact). Model details are explained in the appendix.

\todo{I think we need a section here to describe partnership dynamics/contact structure, i.e. define the $\rho$, $c_u$, $c_e$, $c_w$ parameters - will be similar to \champr}
\input{ms_parms.tex}

\subsection{Latin hypercube sampling}

Despite considerable effort by many researchers \citep[e.g.][]{hollingsworth_hiv1_2008,champredon_hiv_2013}, the parameters determining the rate and structure of sexual partnership change and contact are still very uncertain; this led \champr\ to adopt a Latin hypercube sampling (LHS) strategy \citep{blower_drugs_1991} that evaluates model outcomes over a range of parameter values. In order to make sure that our comparisons among models apply across the entire space of reasonable parameter values, and in order to evaluate the differential sensitivity of different model structures to parameter values, we follow a similar protocol and perform LHS over a parameter set including both the transmission and duration parameters ($\beta_P$, $D_P$, $\beta_D$, $D_D$) and contact/partnership parameters ($\rho$, $c$, $c_u/c_w$, and $c_e/c_w$). We do not allow for uncertainties in parameters that are directly related to the evolutionary process ($\tsub{\beta}{max}$, $\beta_{50}$, $\beta_k$, $\tsub{D}{max}$, $D_{50}$, $D_k$, $\sigma_M$), using \shir's point estimates throughout.

Latin hypercube sampling is done as in \champr. For each parameter, $z$, its range is divided into $N = 1000$ equal intervals on a log scale:

\begin{equation}
z_i = \exp\left(\log(\tsub{z}{min}) + [\log(\tsub{z}{max}) - \log(\tsub{z}{min})] \frac{i-1}{N-1}\right) \qquad i = 1,...,N.
\end{equation}
For simplicity, we assume that all parameters are uniformly distributed on the log scale. Following the vectorization of a parameter range, a matrix is constructed so that each column contains a vector of a parameter series which it represents ($z_1$,...$z_N$). Then, each column is replaced with a random permutation series of itself. Now, each row contains a different parameter set that is used for each simulation run.

Table~\ref{table:parmsTable} gives the ranges of the model parameters used for LHS. Parameter ranges regarding contact and partnership dynamics ($\rho$, $c$, and $c_e/c_w$) are taken from \champr, whereas those regarding infection ($\beta_P$, $D_P$, $\beta_D$, and $D_D$) are taken from \cite{hollingsworth_hiv1_2008}. The remaining parameters are taken from \shir \todo{check: redundant with description a couple of paras above?}

The one completely new parameter in our model, the ratio of uncoupled to within-couple transmission $c_u/c_w$, is needed to more flexibly contrast uncoupled and extra-couple transmission dynamics within multi-strain models (Appendix); we need to pick a reasonable range for it. \champr\ assume that the effective within-couple contact rate and effective uncoupled contact rate have the same range of 0.05 - 0.25.  Given \champr's parameter range, the possible maximum and minimum values of $c_u/c_w$ are 5 and 1/5. Therefore, we use 1/5-5 as the range for the parameter $c_u/c_w$. Although this adds more uncertainty to the parameter $c_u$ --- \champr's range implies a 5-fold difference whereas ours gives a 25-fold difference --- as there is not much known about the uncoupled transmission rate, we consider the wider range appropriate.

\subsection{Simulation runs}

One of the most difficult parts of model comparison is finding parameter sets that are commensurate against many different model structures. For the most part, our models are too complex to easily derive analytical correspondences among them. Given a numerical criterion, such as $r$ (initial exponential growth rate) or $\rzero$ (intrinsic reproductive number), we can adjust one or more parameters by brute force to ensure that all of the models match according to that criterion. While $\rzero$ is often considered the most fundamental property of an epidemic, and might thus seem to be a natural goal, here we focus on matching the initial growth rate $r$ for several reasons. First, our primary interest is in the transient evolutionary dynamics of virulence, which are more strongly affected by $r$ than $\rzero$ \todo{CITE?}. Second, $r$ is in general more directly observable in real epidemics; $r$ can be estimated by simply fitting an exponential curve to the initial incidence or prevalence curves \todo{CITE Ma et al.}, while $\rzero$ typically requires either (1) knowledge of \emph{all} epidemic parameters or (2) relatively sophisticated back-calculation based on $r$ and knowledge of the serial interval or generation interval of the disease \todo{CITE Wallinga, Teunis, Lipsitch, etc.} Thus, we scale a parameter so that every run has the same initial exponential growth rate in the incidence.

In order to allow for all models to have equal initial exponential growth rate, $r$, we need to pick a parameter, $s$, so that $\lim_{s\to 0} r(s) = 0$ and $\lim_{s\to\infty}  r(s) = \infty$. As adjusting either partnership change rate (i.e. partnership formation and dissolution rate) or transmission rate does not fulfill this requirement for certain models, we decided to scale partnership change rate and dissolution rate by the same factor of $\gamma$: $\tsub{\beta}{adj} =  \gamma \tsub{\beta}{base}$, $\tsub{c}{adj} = \gamma \tsub{c}{base}$, $\tsub{\rho}{adj} = \gamma \tsub{\rho}{base}$. Since transmission rate is adjusted by the scale of $\gamma$, uncoupled and extra-couple transmission rates are adjusted as well. For models 3, 4, and 5, all of which assume instantaneous partnership (and hence do not track single individuals), only the transmission rate and partnership dissolution rate (in this case equivalent to the partnership change rate) are adjusted.

We run each model for each of 1000 parameter sets chosen by Latin hypercube sampling \todo{CITE Blower et al - or at first mention of LHS}, with fixed starting conditions
of mean $\log_{10}$ SPVL of 3 and epidemic size of $10^{-4}$. After each run, initial exponential growth rate is calculated. Then, parameters are scaled so that the initial exponential growth rate is scaled to 0.04, which is approximately equal to that of Shirreff et al's model.

For each model we derive the following summary statistics: peak virulence, peak time, equilibrium virulence, and relative peak virulence. The transient phase of an epidemic is often characterized by high virulence, and we define peak virulence as the maximum virulence during this phase. It is simply calculated by taking the maximum value from the virulence trajectory, and peak time is the time at which the maximum value is reached. Once the epidemic enters the endemic phase, evolution of virulence stabilizes and reaches equilibrium. Equilibrium virulence is calculated by taking the mean virulence at 4000 years. Although most simulations reach equilibrium much earlier, we set our time horizon at a much later date as some simulation runs have slow rate of evolution depending on the parameter set and model assumptions.

We focus on these statistics for the following two reasons. First of all, knowing the possible ranges for the peak virulence allows us to estimate the worst-case scenario for the HIV and other sexually transmitted disease epidemics. When an epidemic begins, it is often the case that the pathogen has already evolved towards high virulence by the time it is observed \todo{CITE}. Understanding how virulent a pathogen can evolve before an epidemic begins can be helpful for controlling the disease. Furthermore, knowing the initial virulence, peak virulence, timing of the peak virulence, and equilibrium virulence provide sufficient detail to identify the shape of the virulence trajectory. During an epidemic outbreak, it is difficult to observe virulence evolution. Specifically, in the case of HIV and other sexually transmitted diseases, slow evolutionary time-scale makes observing changes in the mean virulence even more challenging. Knowing the ranges of these statistics can help real-time virulence evolution prediction during an epidemic less troublesome.

\section{Results}

Our simplifications of \shir's model reproduce its qualitative behaviour --- in particular, its predictions of virulence dynamics --- reasonably well. As $r$ decreases from 0.084 to 0.42 (the latter value matching the initial rate of increase in prevalence in \shir's full model) the initial trajectory of increasing virulence brackets the rate from the original model (Figure~\ref{fig:panel3}a). However, our model produces lower peak virulence \todo{fill in these numbers ...} ($\approx ?$ vs. $\approx ?$) and equilibrium virulence $\approx ?$ vs. $\approx?$) than Shirreff's, even for matching initial incidence trajectories (i.e., $r=0.042$).

Changing the initial infectious density ($I(0)$), while it produces the expected changes in the initial epidemic trajectory (Supplementary material), has little effect on the virulence trajectory, making the virulence peaks slightly later and larger as $I(0)$ decreases. Decreasing $I(0)$ allows a longer epidemic phase before the transition to endemic dynamics (Figure~\ref{fig:panel3}b). Decreasing the initial virulence similarly but more strongly leads to progressively later, larger peaks in virulence (Figure~\ref{fig:panel3}c).

\todo{add Shirreff's parameters to table 1: (\sout{Beta1 = 2.76, Duration1 = 0.25}, Beta3 = 0.76, Duration3 = 0.75, \sout{cmeanbase = 1.25, alphaDist = c(min = 2, max = 7,  delta = 0.1)}).}

<<setup2,echo=FALSE>>=
orig_sum_labs <- c("peak_time","peak_vir","eq_vir","rel_peak")
new_sum_labs <- c("peak time","peak virulence","equil virulence","relative peak")
fixfac2 <- function(x) factor(x,
                       levels=orig_sum_labs,
                       labels=new_sum_labs)
@

<<panel3,fig.cap="Baseline dynamics. Time series of mean population virulence ($\\log_{10}$ SPVL). (a) Shirreff model, effects of varying $r$. (b) Effects of varying initial infectious density $I(0)$. (c) Effects of varying initial mean virulence $\\alpha(0)$. Parameters",dev="tikz",fig.height=3,fig.width=8,cache=TRUE>>=
for (i in c("","2","3")) 
    load(sprintf("../simdata/ev_LHS_resS%s.rda",i))
Ls <- load("../simdata/shirreff_data.rda")
tvec <- seq(from=1,by = 0.1,length.out=nrow(vir_matS))
dimnames(vir_matS) <- list(Time=tvec,
                           run=paste0("$r=",c(0.021,0.042,0.084),"$"))

d_littleR_m <- melt(vir_matS,value.name="Mean.VL")
shirreff_df_m <- subset(res_sum_df,run==1,select=c(run,Time, Mean.VL))
shirreff_df_m$run <- "Shirreff"
df_tot <- rbind(shirreff_df_m,d_littleR_m)
vlab <- "mean virulence ($\\log_{10}$ SPVL)"
tlab <- "time (years)"
g1 <- ggplot(df_tot,aes(x=Time,y=Mean.VL,colour=run))+
    geom_line()+labs(x=tlab,y=vlab)+
    theme(legend.position=c(0.8,0.2))
###
dimnames(vir_matS2) <- list(Time=tvec,
               run=c("$10^{-3}$", "$10^{-4}$", "$10^{-5}$"))
g2 <- g1 %+% melt(vir_matS2,id.var="Time",value.name="Mean.VL")
###
dimnames(vir_matS3) <- list(Time=tvec,
                            run=c("$2.5$", "$3$", "$3.5$"))
g3 <- g1 %+% melt(vir_matS3,id.var="Time",value.name="Mean.VL")
limits <- c(2.5,4.7)
breaks <- seq(limits[1],limits[2], by = 0.5)
afun <- function(label,size=8) {
    annotate(geom="text",label=label,x=-Inf,y=Inf,
             size=size,vjust=1,hjust=0)
}
g1.y <- g1 + scale_y_continuous(limits = limits, breaks = breaks)+
    scale_colour_discrete(name="run")+afun("a")
g2.y <- g2 + scale_y_continuous(limits = limits, breaks = breaks)+
    scale_colour_discrete(name="$I(0)$")+afun("b")
g3.y <- g3 + scale_y_continuous(limits = limits, breaks = breaks)+
        scale_colour_discrete(name="$\\alpha(0)$")+afun("c")
grid.arrange(g1.y, g2.y, g3.y, nrow=1)
@ 

Across the entire range of parameters covered by the LHS analysis, all of the classes of models we considered produce qualitatively similar virulence trajectories (Figure~\ref{fig:virtraj}). Although the speed of virulence evolution varies, leading to wide variation in the peak virulence (means ranging from approximately 3.75 to 4.5 $\log_{10}$ SPVL), virulence peaks in all models between 200 and 300 years.

<<virtraj,fig.cap="Envelopes of virulence trajectories under all models",fig.width=6,cache=TRUE,message=FALSE,warning=FALSE,dev="tikz">>=
gg_virtraj <- ggplot(sum_list[["vir_mat"]],
                     aes(tvec,mean,ymin=lwr,ymax=upr))+
    geom_line(aes(colour=model,linetype=model),lwd=1.5)+
    geom_ribbon(aes(fill=model),alpha=0.3)+
    labs(x="time",y=vlab)+
    scale_x_continuous(limits=c(0,2500))
print(gg_virtraj)
@ 

<<unidist,fig.cap="Univariate distributions of summary statistics. Note: point for eqvir/random is a placeholder, will have to be explained",fig.width=8,fig.height=4.8>>=
sL <- transform(sum_list[["sum_mat"]],rel_peak=peak_vir/eq_vir)
mL <- melt(sL,id.vars=c("model","run"))
## horrible hack (but doesn't help); subsample
## w <- which(mL$model=="random")
## mLw <- mL[-sample(w,size=length(w)*9/10,replace=FALSE),]
w <- with(mL,which(model=="random" & variable=="eq_vir"))
rval <- mean(mL$value[w])
mLw <- droplevels(mL[-w,])
mLw$variable <- fixfac2(mLw$variable)
gg_univ <- ggplot(mLw,aes(value,model,fill=model))+
    geom_violinh(width=1)+
    facet_wrap(~variable,scale="free_x")+
    guides(fill=guide_legend(reverse=TRUE))+
    labs(y="")+
    geom_point(data=data.frame(model="random",variable="equil virulence",
                               value=rval),pch=22,size=3,show.legend=FALSE)
print(gg_univ)
@

<<pairplot,echo=FALSE,fig.cap="pairs plot",fig.width=7,fig.height=7>>=
sL2 <- sL
names(sL2)[na.omit(match(orig_sum_labs,names(sL)))] <- gsub(" ","_",new_sum_labs)
    ## paste0("`",new_sum_labs,"`")
ggp1 <- ggpairs(sL2,
        mapping = ggplot2::aes(color = model),
        columns=3:6,
        lower = list(continuous = wrap("points", alpha = 0.5,size=0.5)),
        diag = list(continuous = "blankDiag"),
        upper = list(continuous = "blank"))+
    theme(panel.margin=grid::unit(0,"lines"))
## http://stackoverflow.com/questions/14711550/is-there-a-way-to-change-the-color-palette-for-ggallyggpairs-using-ggplot
## have to change plot one panel at a time
cbscales <- list(scale_colour_brewer(palette = 'Dark2'),
                 scale_fill_brewer(palette = 'Dark2'))  ## DRY
ggp2 <- ggp1
for (row in seq_len(ggp2$nrow))
    for (col in seq_len(ggp2$ncol))
        ggp2[row, col] <- ggp2[row, col] + cbscales[[(row==col)+1]]
## hack to remove top row/right column
ggp3 <- ggp2
ggp3$nrow <- ggp3$ncol <- 3
ggp3$plots <- ggp3$plots[c(5:7,9:11,13:15)]
ggp3$xAxisLabels <- ggp3$xAxisLabels[1:3]
ggp3$yAxisLabels <- ggp3$yAxisLabels[2:4]
print(ggp3)
@

Our chosen summary statistics (peak time, peak virulence, equilibrium
virulence, and relative peak virulence) all vary considerably across models.
We first consider the models of intermediate realism (implicit,
instantaneous-switching with and without extra-pair contact, and
pair formation without extra-pair contact). Some parameter
sets for these models lead to low equilibrium virulence ($\approx 2.5 \log_{10} SPVL$)
these same sets lead to correspondingly low
peak virulence ($< 3.5 \log_{10} SPVL$) and early peak times (before 200 years), 
but high relative peaks ($>1.3$)
(Figure~\ref{fig:pairplot}, leftmost column) because the equilibrium virulence is low.
At the opposite extreme, parameter sets that produce high equilibrium virulence 
also produce late peaks ($> 200 \text{years}$), 
high peak virulence, and low relative peaks ($\approx 1.05$).
The pair-formation without extra-pair contact and implicit models
occasionally have parameter sets that select for such low virulence across
the board that they never exceed their initial virulence, leading to a tail
of peak times near 0.

instswitch is intermediate beween (pairform + implicit) 

<<testrev,echo=FALSE,eval=FALSE>>=
set.seed(101)
zz <- data.frame(x=rnorm(3000),f=factor(rep(letters[1:3],each=1000)))
g0 <-  ggplot(zz,aes(x=f,y=x,fill=f))+geom_violin()
grid.arrange(
    g0,
    g0+ guides(fill=guide_legend(reverse=TRUE)),
    nrow=2)
@


<<plot_sens,dev="png",echo=FALSE,fig.width=10,fig.height=6>>=
ggplot(mL2,aes(LHSval,sumval,colour=model))+
    geom_point(pch=".",alpha=0.5)+
    facet_grid(sumvar~LHSvar,scales="free")+
    geom_smooth(se=FALSE)+labs(x="",y="")+
    scale_x_log10()
@

Figure to-do: maybe go back to bw theme?
\begin{itemize}
\item fig 1. tweak size? remove redundant y axes/labels?
\item fig 2. 6-panel figure? log scale? boundary lines instead of ribbons for edges of envelopes? move labels away from axes?
\item fig 3. change order (Random after pairform)
\item fig 4. add legend, fix axis labels; maybe transparent
\item fig 5. sensitivity plot (unscaled plot); take out plots that are not related ... implicit model doesn't use rho, cu or ce; random doesn't use rho, cmean?, cu, ce; instswitch doesn't have rho or cu or ce; models without epc don't have ce or cu; instswitch + epc doesn't have rho and cu (no uncoupled compartment). Maybe squash panels together, higher resolution

\end{itemize}


\section{Discussion}

\section{To do}
\begin{itemize}
\item clean up code and results
\item incorporate pictures
\item finish writing!
\item hard parts of the results
\begin{itemize}
\item explaining similarities among the models: why does implicit look most like pair formation model? (see R0comparison HTML)
\item explaining sensitivity plot
\end{itemize}
\end{itemize}

\appendix

See \href{http://tinyurl.com/bolker-jrsi-cites}{Bolker JRSI Google scholar cites} (\href{https://scholar.google.ca/scholar?oi=bibs&hl=en&cites=18202064200267310847}{alternative link})

\section{Model details}

Since we use multi-strain models, in which the distribution of $\log_{10}$ SPVL has been discretized into a vector, we use a matrix notation to describe our models. Five states described in the method section is replaced with the following notations: $S$, $I_i$, $SS$, $SI_i$, $II_{ij}$. Subscript indicates a strain that an individual is infected with. For example, $I_i$ is number of infected individuals with $\log_{10}$ SPVL of $\alpha_i$, and $II_{ij}$ is the number of concordant couples, in which two partners have $\log_{10}$ SPVL of $\alpha_i$ and $\alpha_j$. $II_{ij}$ is equivalent to $II_{ji}$. Lastly, we define Kronecker delta as follows: 

\begin{equation}
\delta_{ij} = \left.\begin{aligned}
0 & \quad \text{if} \quad i \neq j,\\
1 & \quad \text{if} \quad i = j.
\end{aligned}\right\rbrace
\end{equation}

In our models, we introduce a non-standard use of Kronecker delta as exponent, such as $2^{\delta_{ij}}$. In a single-strain model, both individuals in a concordant partnership, $II$, enter single-infected compartment, $I$, when a partnership dissolves: $I' = 2 c II'$. However, this is not the case in a multi-strain model. If infected individuals in an $II$ partnership have different strains, they will enter different compartment. On the other hand, if both partners in an $II$ partnership are infected with same strains, both will enter the same compartment. Therefore, we use Kronecker delta as exponent to distinguish such difference. Further details are explain below.

\todo{explain non-standard use of Kronecker delta as exponent!}

\subsection*{Model 1 and 2 - Partnership dynamic}

Single individuals acquire partners at rate $\rho$: $S' = - \rho S$ and $I_i' = - \rho I_i$. We follow Champredon et al's results and assume that single individuals are distributed into couple states through binomial distribution:


\begin{equation}
\begin{aligned}
SS' &= \frac{\rho S \cdot S}{2 (S + \sum_k I_k)}\\
SI_i' &= \frac{\rho S \cdot I_i}{S + \sum_k I_k}\\
II_{ij}' &= \left(\frac{1}{2}\right)^{\delta_{ij}} \cdot \frac{\rho I_i \cdot I_j}{S + \sum_k I_k}
\end{aligned}
\end{equation}
We introduce the Kronecker delta above to differentiate the partnership formation rate for $II_{ij}$ 
when $i = j$ from that of $I \neq j$. When $i = j$ --- e.g. in the $II_{ij}$ term --- the partnership formation rate becomes $II_{ii}' = \frac{\rho I_i \cdot I_i}{2(S + \sum_k I_k)}$ due to binomial distribution \todo{??}. On the other hand, when $i \neq j$, the partnership rate is $II_{ij}' = \frac{\rho I_i I_j}{S + \sum_k I_k}$.

Partnerships dissolve at rate $c$: $SS' = -c SS$, $SI_i' = -c SI_i$, and $II_{ij} = - c II_{ij}$. Unlike single strain model, where both individuals leaving the $II$ partnership would enter $I$, we have to account for strains which the individuals in concordant partnership are infected with (i.e. both partners in $II_{ii}$ enter $I_i$ whereas only one partner in $II_{ij}$ enters $I_i$).

\begin{equation}
\begin{aligned}
S' &= 2 c SS + \sum_k c SI_k \\
I_i' &= c SI_i + \sum_k 2^{\delta_{ik}} c II_{ik}\\
\end{aligned}
\end{equation}
Combining partnership formation and dissolution process yields the following equation:

\begin{equation}
\begin{aligned}
S' &= - \rho S + 2 c SS + \sum_k c SI_k \\
I_i' &= - \rho I_i + c SI_i + \sum_k 2^{\delta_{ik}} c II_{ik}\\
SS' &= \frac{\rho S \cdot S}{2 (S + \sum_k I_k)} - c SS\\
SI_i' &= \frac{\rho S \cdot I_i}{S + \sum_k I_k} - c SI_i\\
II_{ij}' &= (\frac{1}{2})^{\delta_{ij}} \cdot \frac{\rho I_i \cdot I_j}{S + \sum_k I_k} - c II_{ij}
\end{aligned}
\end{equation}

\subsection*{Model 1 and 2 - Infection}

Within-couple transmission occurs in both models. An infected partner in $SI$ partnership transmits virus to a susceptible partner, and partnership state becomes $II$: $SI_i' = - \beta_i SI_i$. Since we assume that mutation occurs, $II_{ij}$, where $i \neq j$, can be formed from both $SI_i$ and $SI_j$ partnership: $II_{ij}' = M_{ij} \beta_i SI_i + M_{ji} \beta_j SI_j$. On the other hand, $II_{ii}$ can only be formed from an $SI_i$ partnership: $II_{ii}' = M_{ii} \beta_i SI_i$. Using the Kronecker delta notation, we obtain following set of equations that describe within-couple transmission dynamics:

\begin{equation}
\begin{aligned}
SI_i' &= - \beta_i SI_i\\
II_{ij} &=  (\frac{1}{2})^{\delta_{ij}} \cdot (M_{ij} \beta_i SI_i + M_{ji} \beta_j SI_j)
\end{aligned}
\end{equation}

Champredon et al define the proportion of infectious extra-couple and uncoupled contact through the following term:

\begin{equation}
P = \frac{c_u I + c_e (SI + 2 II)}{c_u (S + I) + 2 c_e(SS + SI + II)}.
\end{equation}
Effective uncoupled, $c_u$, and extra couple, $c_e$, contact rate can be divided into two terms: uncoupled/extra-couple contact rate $\times$ rate of transmission per contact. Therefore, transmission rate per contact term in $c_u$ and $c_e$ is canceled out in the equation above. Using this property, we modify the equation above as follows:

\begin{equation}
P = \frac{r_u I + r_e (SI + 2 II)}{r_u (S + I) + 2 r_e(SS + SI + II)},
\end{equation}
where $r_u = c_u/c_w$ and $r_e = c_e/c_w$ are the relative uncoupled/extra-couple contact rates. This simplification is useful in a multi-strain model since we cannot multiply a vector with a single value (e.g. $c_u S$ in denominator) if we use Champredon et al's equation as it is. Extending the above equation to multi-strain model so that $P_i$ represents the proportion of the extra-couple and uncoupled contact of an infected individual with strain $i$, we obtain the following equation:

\begin{equation}
P_i = \frac{r_u I_i + r_e (SI_i + \sum_k (II_{ik} + \delta_{ik} II_{ik}))}{r_u (S + \sum_k I_k) + r_e(2 SS + \sum_k 2 SI_k + \sum_l \sum_k (1 + \delta_{lk})II_{lk} )},
\end{equation}
Using the equation above, we can model uncoupled and extra-couple mixing. For convenience, uncoupled and extra-couple transmission rates, $c_u$ and $c_e$, will be replaced with $U_i = r_u \beta_i$ and $E_i = r_e \beta_i$ from now on.

Single susceptible individuals become infected and enter single infected compartment at the total rate of $\sum_k P_k U_k S$. Through mutation, newly infected individuals are distributed into each single infected compartments with different strains: $I_i' = \sum_k M_{ki} P_k U_k S$. Either partners in $SS$ partnership can be infected and the partnership state can become $SI$ partnership at the total rate of $\sum_i 2 P_i E_i SS$. Formation of $SI_i$ partnership is similar to the process through which single susceptible individuals are distributed into single infected compartments: $SI_i' = \sum_k 2 M_{ki} P_k E_k SS$. Lastly, susceptible partner of an $SI$ partnership can be infected due to uncoupled/extra-couple contacts and partnership can move to an $II$ partnership. Like previous cases, $SI_i$ partnership moves out of the compartment at a rate of $\sum_k P_k E_k SI_i$. Mutation process is similar to that of within-couple transmission. The only difference is that the $\log_10$ SPVL of a newly infected partner is not determined by its original partner but from an extra couple partner (i.e. the term $P_i$): $II_{ij}' = (1 -\frac{\delta_{ij}}{2})(\sum_k (M_{kj} P_k E_k SI_i + M_{ki} P_k E_k SI_j))$. Combining these equations we get the following set of equations that describe all the transmission dynamics

\subsection*{Model 1 and 2 - Disease induced mortality}

Disease induced death rate, $\lambda$, is given by taking the reciprocate of the total duration of the infection: $\lambda_i = 1/(D_A + D_P(\alpha_i) + D_D)$. Since we assume SIS formulation, where infected individuals that die from infection enter single susceptible compartment, we obtain the following equation for the single infected individuals:

\begin{equation}
\begin{aligned}
S' &= \sum_k \lambda_k I_k \\
I_i &= - \lambda_i I_i \\
\end{aligned}
\end{equation}

If an infected individual in a partnership dies, partnership dissolves. Thus, an $SI_i$ partnership dissolves at a rate $-\lambda_i$, and the susceptible partner enters single susceptible compartment at rate $\lambda_i SI_i$ (infected partner that dies enter single susceptible compartment at an equal rate as well due to SIS formation):

\begin{equation}
\begin{aligned}
S' &= \sum_k 2 \lambda_k SI_k \\
SI_i &= - \lambda_i SI_i 
\end{aligned}
\end{equation}

Similarly, $II_{ij}$ partnership dissolves at a rate $-(\lambda_i + \lambda_j)$, but two cases, when $i \neq j$ and $i = j$, must be considered separately. When an $II_{ij}$ partnership dissolves due to disease induced mortality, where $i \neq j$, death of partner with strain $i$ causes its partner to enter $I_j$ compartment at rate $\lambda_j II_{ij}$, and vice versa. When an $II_ii$ partnership dissolves, death of either partner causes the other partner to enter $I_i$ compartment at rate $\lambda_i II_{ii}$, which sums up to $2\lambda_i II_{ii}$. Combining these dynamics yield the following equation:

\begin{equation}
\begin{aligned}
S' &= \sum_l \sum_k  2^{\delta_{lk}} \lambda_k II_{lk} \\
I_i' &=  \sum_k 2^{\delta_{ik}} \lambda_k II_{ik} \\
II_{ij}' &= -(\lambda_i + \lambda_j) II_{ij}
\end{aligned}
\end{equation}

Finally, combining all these equations give us the full model, which is model 1. We can simply take out the uncoupled and extra couple transmission term to obtain equation 2:

\begin{equation}
\begin{aligned}
S' =& - \rho S + 2 c SS + \sum_k c SI_k - \sum_k P_k U_k S + \sum_k \lambda_k I_k \\
&+ \sum_k 2 \lambda_k SI_k + \sum_l \sum_k  2^{\delta_{lk}} \lambda_k II_{lk}\\
I_i' =&  - \rho I_i + c SI_i + \sum_k 2^{\delta_{ik}} c  II_{ik} + \sum_k M_{ki} P_k U_k S- \lambda_i I_i \\
&+ \sum_k 2^{\delta_{ik}} \lambda_k II_{ik} \\
SS' =& \frac{\rho S \cdot S}{2 (S + \sum_k I_k)} - c SS - \sum_i 2 P_i E_i SS \\
SI_i' =& \frac{\rho S \cdot I_i}{S + \sum_k I_k} - c SI_i - \beta_i SI_i + \sum_k 2 M_{ki} P_k E_k SS - \sum_k P_k E_k SI_i   \\
&- \lambda_i SI_i\\
II_{ij}' =& (\frac{1}{2})^{\delta_{ij}} \cdot \frac{\rho I_i \cdot I_j}{(S + \sum_k I_k)} - c II_{ij} + (\frac{1}{2})^{\delta_{ij}} \cdot (M_{ij} \beta_i SI_i + M_{ji} \beta_j SI_j) \\
&+ (\frac{1}{2})^{\delta_{ij}} \cdot (\sum_k (M_{kj} P_k E_k SI_i + M_{ki} P_k E_k SI_j)) -(\lambda_i + \lambda_j) II_{ij}
\end{aligned}
\end{equation}

\subsection*{Model 3 and 4 - Partnership dynamic}

Since model 3 and 4 assume instantaneous partnership formation, there are only three states: $SS$, $SI_i$, and $II_{ij}$. Partnership dissolution is equal to that of model 1 and 2: $SS' = -cSS$, $SI_i' = - cSI_i$, and $II_{ij}' = - II_{ij}$. Once the individuals leave partnership, they enter temporary compartments and are distributed into a partnership through binomial distribution:

\begin{equation}
\begin{aligned}
X &= 2 c SS + \sum_k c SI_k \\
Y_i &= c SI_i + \sum_k 2^{\delta_{ik}} c II_{ik} \\
SS' &= - c SS + \frac{X^2}{2 (X + \sum_k Y_k)}\\
SI_i' &= - c SI_i + \frac{X Y_i}{X + \sum_k Y_k}\\
II_{ij}' &= - c II_{ij} +(\frac{1}{2})^{\delta_{ij}} \frac{Y_i Y_j}{X + \sum_k Y_k}.
\end{aligned}
\end{equation}

\subsection*{Model 3 and 4 - Infection}

Model 3 and 4 share the within-couple transmission term with model 1 and 2. Since there is no single state, only extra couple transmission exists:

\begin{equation}
P_i = \frac{r_e (SI_i + \sum_k (II_{ik} + \delta_{ik} II_{ik}))}{r_e(2 SS + \sum_k 2 SI_k + \sum_l \sum_k (II_{lk} + \delta_{kl} II_{lk}) )}.
\end{equation}
Movement from $SS$ state to $SI$ state and $SI$ to $SS$ is modeled through the same equation that is used in model 1 and 2.

\subsection*{Model 3 and 4 - Disease induced mortality}

Disease induced mortality is modeled similar to model 1 and 2. However, as single state does not exist in model 3 and 4, individuals that has left their partnerships due to death of their partners enter temporary compartments and form partners instantly:

\begin{equation}
\begin{aligned}
X &= \sum_k 2 \lambda_k SI_k + \sum_l \sum_k 2^{\delta_{lk}}  \lambda_k II_{lk} \\
Y_i &=  \sum_k  2^{\delta_{ik}}  \lambda_k II_{ik} \\
SS = &= \frac{X^2}{2 (X + \sum_k Y_k)}\\
SI_i &= - \lambda_i SI_i + \frac{X Y_i}{X + \sum_k Y_k}\\
II_{ij}' &= -(\lambda_i + \lambda_j) II_{ij} + (\frac{1}{2})^{\delta_{ij}} \cdot \frac{Y_i Y_j}{X + \sum_k Y_k}
\end{aligned}
\end{equation}

Combining all these dynamics, we have equation 3. If we remove extra-couple transmission, we have equation 4.

\begin{equation}
\begin{aligned}
X =& 2 c SS + \sum_k c SI_k + \sum_k 2 \lambda_k SI_k + \sum_l \sum_k 2^{\delta_{lk}}  \lambda_k II_{lk}\\
Y_i =& c SI_i + \sum_k 2^{\delta_{ik}}  c II_{ik} + \sum_k  2^{\delta_{ik}}  \lambda_k II_{ik} \\
SS  =& - c SS + \frac{X^2}{2 (X + \sum_k Y_k)}  - \sum_i 2 P_i E_i SS\\
SI_i =& - c SI_i + \frac{X Y_i}{X + \sum_k Y_k} - \beta_i SI_i + \sum_k 2 M_{ki} P_k E_k SS\\
&- \sum_k P_k E_k SI_i - \lambda_i SI_i  \\
II_{ij}& - c II_{ij} +(\frac{1}{2})^{\delta_{ij}} \frac{Y_i Y_j}{X + \sum_k Y_k} + (\frac{1}{2})^{\delta_{ij}} \cdot (M_{ij} \beta_i SI_i + M_{ji} \beta_j SI_j) \\
&+ (\frac{1}{2})^{\delta_{ij}} \cdot (\sum_k (M_{kj} P_k E_k SI_i + M_{ki} P_k E_k SI_j)) -(\lambda_i + \lambda_j) II_{ij}
\end{aligned}
\end{equation}

\subsection*{Model 5}

Model 5 is an implicit instantaneous partnership formation model, which uses adjusted transmission rate, $\beta'$, that is derived from Hollingsworth et al's approximated basic reproduction number:

\begin{equation}
\beta'_i = \frac{c \beta_i}{c + \beta_i + \lambda_i}.
\end{equation}
Thus, we get the following model:

\begin{equation}
\begin{aligned}
S' & = \sum_k \lambda_k I_k - \sum_k \beta'_k S I_k\\
I_i & = \sum_k M_{ki} \beta'_k S I_k - \lambda_i I_i
\end{aligned}
\end{equation}

\subsection*{Model 6}

Model 6 is a random mixing model. It is modeled in a same way as model 6 without the adjusted transmission rate:

\begin{equation}
\begin{aligned}
S' & = \sum_k \lambda_k I_k - \sum_k \beta_k S I_k\\
I_i & = \sum_k M_{ki} \beta_k S I_k - \lambda_i I_i
\end{aligned}
\end{equation}


\subsection*{Initial distribution of infected individuals}

We follow Champredon et al's result to calculate the initial distribution of infected individuals. For model 1 and 2, we have disease equilibrium state of $S^* = \frac{c}{c + \rho}$ and $SS^* = \frac{1-S^*}{2}$. We let $\epsilon = 10^{-4}$, which is the total number of infected individuals in the beginning of simulation and $D$ be the vector such that $D_i$ represent the proportion of individuals with $\log_{10}$ SPVL of $i$. $Y_i$ is taken from normal distribution with mean 3 and is normalized so that $\sum_i D_i = 1$. Then, we have the following initial distribution of each states:

\begin{equation}
\begin{aligned}
S(0) &= (1 - \epsilon) S^* \\
SS(0) &= (1 - \epsilon)^2 SS^*\\
SI_i(0) &= 2 \epsilon (1-\epsilon) SS^* D_i\\
I_i(0) &=  \epsilon S^* D_i\\
II_{ij}(0) &=  (\frac{1}{2})^{\delta_{ij}} 2\epsilon^2 SS^* D_i D_j.
\end{aligned}
\end{equation}
Since model 3 and 4 do not have single state, $SS^*=1$ at the disease free equilibrium and the initial distribution becomes as follows:

\begin{equation}
\begin{aligned}
SS(0) &= (1 - \epsilon)^2 SS^*\\
SI_i(0) &= 2 \epsilon (1-\epsilon) SS^* D_i\\
II_{ij}(0) &=  (\frac{1}{2})^{\delta_{ij}} 2\epsilon^2 SS^* D_i D_j.
\end{aligned}
\end{equation}
Lastly, as model 5 is an implicit model, which does not consider different stages of partnership, we have the following initial distribution.

\begin{equation}
\begin{aligned}
S(0) &= 1 - \epsilon\\
I_i(0) &=  \epsilon D_i.
\end{aligned}
\end{equation}
Model 6 also shares the same distribution of initial infected individuals as model 5.

\bibliographystyle{chicago}
\bibliography{virulence}
\end{document}


